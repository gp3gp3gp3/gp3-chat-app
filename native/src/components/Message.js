import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'
import moment from 'moment'

class Message extends PureComponent {
  constructor (props) {
    super(props)
    this.state = {
      messageTime: moment(props.time).fromNow()
    }
    this.updateMessageTime = this.updateMessageTime.bind(this)
  }

  componentDidMount () {
    this.interval = setInterval(this.updateMessageTime, 3000)
  }

  componentWillUnmount () {
    clearInterval(this.interval)
  }

  updateMessageTime () {
    const messageTime = moment(this.props.time).fromNow()
    this.setState({ messageTime })
  }

  render () {
    const {
      user: { displayName, uid },
      message,
      currentUser: { uid: currentUserId }
    } = this.props
    const isCurrentUser = uid === currentUserId
    return (
      <View className='Message-wrapper'>
        {!isCurrentUser && <Text
          className={`Message-time ${isCurrentUser ? 'current-user' : 'other-user'}`}
        >{displayName}</Text>}
        <Text
          className={`Message ${isCurrentUser ? 'current-user blue' : 'other-user green'}`}
        >{message}</Text>
        <Text
          className={`Message-time ${isCurrentUser ? 'current-user' : 'other-user'}`}
        >{this.state.messageTime}</Text>
      </View>
    )
  }
}

export default Message
