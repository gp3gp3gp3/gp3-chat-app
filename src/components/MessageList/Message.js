import React from 'react'
import './MessageList.css'

const Message = ({ message }) => (
  <span className='Message'>{message}</span>
)

export default Message
