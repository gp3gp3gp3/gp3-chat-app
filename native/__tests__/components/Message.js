/* eslint-env jest */
import 'react-native'
import { shallow } from 'enzyme'
import React from 'react'
import Message from '../../src/components/Message'

it('renders correctly', () => {
  const props = {
    user: {
      displayName: 'Test User',
      uid: '1234'
    },
    currentUser: {
      uid: '123'
    }
  }
  const wrapper = shallow(
    <Message
      {...props}
    />
)
  expect(wrapper).toMatchSnapshot()
})
