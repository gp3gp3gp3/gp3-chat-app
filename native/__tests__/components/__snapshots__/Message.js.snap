// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    currentUser={
      Object {
        "uid": "123",
      }
    }
    user={
      Object {
        "displayName": "Test User",
        "uid": "1234",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          className="Message-time other-user"
          ellipsizeMode="tail"
        >
          Test User
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          className="Message other-user green"
          ellipsizeMode="tail"
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          className="Message-time other-user"
          ellipsizeMode="tail"
        >
          a few seconds ago
        </Text>,
      ],
      "className": "Message-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Test User",
          "className": "Message-time other-user",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Test User",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "className": "Message other-user green",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "a few seconds ago",
          "className": "Message-time other-user",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "a few seconds ago",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            className="Message-time other-user"
            ellipsizeMode="tail"
          >
            Test User
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            className="Message other-user green"
            ellipsizeMode="tail"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            className="Message-time other-user"
            ellipsizeMode="tail"
          >
            a few seconds ago
          </Text>,
        ],
        "className": "Message-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Test User",
            "className": "Message-time other-user",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Test User",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "className": "Message other-user green",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "a few seconds ago",
            "className": "Message-time other-user",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "a few seconds ago",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
